cmd/nitro/init.go

```diff
@@ -184,6 +186,7 @@ func validateBlockChain(blockChain *core.BlockChain, expectedChainId *big.Int) e
 	if chainId.Cmp(expectedChainId) != 0 {
 		return fmt.Errorf("attempted to launch node with chain ID %v on ArbOS state with chain ID %v", expectedChainId, chainId)
 	}
+	// TODO should we validate also chainConfig?
```

Comment:
Hmm, yeah it seems logical to require that the blockchain's `Config()` is compatible with the ArbOS stored chain config (if it exists) as of the **latest** block number and block timestamp (get that from `CurrentBlock()` not `CurrentHeader()` the latter is out of sync with state sometimes for complicated reasons). We can check that it's compatible with the latest one because we're also pulling state from the latest block.